hepdata/ext/elasticsearch/api.py

```diff
@@ -497,3 +497,27 @@ def get_count_for_collection(doc_type, index=None):
     :return: the number of records in that collection
     """
     return es.count(index=index, q='doc_type:'+doc_type)
+
+
+@default_index
+def get_all_ids(index=None, id_field='recid', last_updated=None, latest_first=False):
+    """Get all record or inspire ids of publications in the search index
+
+    :param index: name of index to use.
+    :param id_field: elasticsearch field to return. Should be 'recid' or 'inspire_id'
+    :return: list of integer ids
+    """
+    if id_field not in ('recid', 'inspire_id'):
+        raise ValueError('Invalid ID field %s' % id_field)
+
+    search = Search(using=es, index=index) \
+        .filter("term", doc_type=CFG_PUB_TYPE) \
+        .source(fields=[id_field])
+
+    if last_updated:
+        search = search.filter("range", **{'last_updated': {'gte': last_updated.isoformat()}})
+
+    if latest_first:
+        search = search.sort({'last_updated' : {'order' : 'desc'}}).params(preserve_order=True)
+
+    return [h[id_field] for h in search.scan()]
```

Comment:
I took the cast out because the default data in the tests ends up storing integer values in ES for the inspire IDs, which I hadn't realised until now is not consistent with the normal behaviour. Will investigate further.

Reply:
See #301 for test data issue.